### PR TITLE
Cu 86bwmcp3c/build my goal component

### PR DESCRIPTION
**Pull request by Elyzsium**
**Description of PR**

updating goal component 

---

Please check all that apply.

#### Checklist:

- [x] Rebased with master
- [x] PR Description provided
- [ ] Did you communicate these changes with the affected teams?
- [ ] Did unit tests/integration tests/E2E tests get properly added/updated?
- [ ] Is this change production-ready including testing, feature toggles, configurations, and deployment considerations?

#### Frontend Checklist:

- [ ] Is this responsive and mobile friendly?
- [ ] Did you follow the [ARIA specs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

#### Backend Checklist:

- [ ] Does the endpoint return the correct success/error format?

---

**Screenshots*
my goals

---

**Test Coverage**  
_add relevant screenshots here_

### CODE REVIEW

**Comment by KristofaJosh** [comment]
Task linked: [CU-86bwmcp3c Build my goal component](https://app.clickup.com/t/86bwmcp3c)

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+// Import statements
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:1]: There's no need for this

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import { Card, Text } from 'components';
+
+
+const data = [
+  {
+    id: '1',
+    image: Travel,
+    target: 'Travel',
+    amount_saved: 5_000_000.0,
+    target_amount: 10_000_000.0,
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:16]: why did you this?

we should use numbers only

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    target: 'Dream Car',
+    amount_saved: 1_000_000.0,
+    target_amount: 5_000_000.0,
+  },
+];
+const MyGoals = () => {
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <span className={styles.headerText}>MY GOALS</span>
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:37]: you can use the Typography element here

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        <Link to="/dashboard/savings" className={styles.view}>
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:38]: use this instead:` routes.dashboard.savings.abs_path`

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    target: 'Dream Home',
+    amount_saved: 10_000_000.0,
+    target_amount: 40_000_000.0,
+  },
+  {
+    id: '3',
+    image: Car,
+    target: 'Dream Car',
+    amount_saved: 1_000_000.0,
+    target_amount: 5_000_000.0,
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:30]: use camelCase

File: apps/client/src/components/compounds/goal/MyGoals.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                    .replace(',', '')
+                ) /
+                  parseFloat(
+                    target.target_amount
+                      .toString()
+                      .replace('₦', '')
+                      .replace(',', '')
+                  )) *
+                  100
+              )}
```
Comment (KristofaJosh) [inline apps/client/src/components/compounds/goal/MyGoals.tsx:84]: can be moved to a function

**Review by KristofaJosh — CHANGES_REQUESTED** [review]
Update your code accordingly

**Review by KristofaJosh — APPROVED** [review]
I believe we can improve the calculate percentage function

**Review by AnosikeChidinma — APPROVED** [review]

**Comment by Dallat65** [comment]
Welldone! Looks nice.

**Review by Dallat65 — APPROVED** [review]

**Pull request merged**